pkg/util/k8sversion_test.go

```diff
@@ -0,0 +1,78 @@
+package util
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/version"
+)
+
+type testMetadata struct {
+	comment        string
+	expectedMinor  int
+	expectedPretty string
+	version        *version.Info
+}
+
+var testMetadatas = []testMetadata{
```

Comment:
I just curious, how do you choose the test cases. When I see the release, the logic should work. 
https://github.com/kubernetes/kubernetes/releases
However, You assume `1.18+` or `1.19.84324.2`. Where do you find the cases?

Reply:
@TsuyoshiUshio similar version string is returned by from the cluster in the `version.Info{}` struct.
For example:
- as reported in #1176, minikube returns string `1.19`
- OpenShift cluster I am using returns string `1.18+`
etc

We only care about the minor version (and specifically if it is > 17), so I added a couple of scenarios to check we can always parse the minor version.

Reply:
Thank you for sharing!